lib/Dialect/Triton/IR/Ops.cpp

```diff
@@ -449,8 +449,8 @@ mlir::LogicalResult mlir::triton::DotOp::inferReturnTypes(
 }
 
 LogicalResult mlir::triton::DotOp::verify() {
-  auto aTy = getOperand(0).getType().cast<RankedTensorType>();
-  auto bTy = getOperand(1).getType().cast<RankedTensorType>();
+  auto aTy = getA().getType();
+  auto bTy = getB().getType();
```

Comment:
Just a nit, but upstream LLVM would want to spell the type instead of `auto`. 
You may decide to consider it "obvious" for a triton developer though.